src/main/java/io/fabric8/elasticsearch/plugin/PluginClient.java

```diff
@@ -53,6 +70,40 @@ public boolean indexExists(final String index) {
         LOGGER.trace("Index '{}' exists? {}", index, exists);
         return exists;
     }
+    
+    public boolean documentExists(final String index, final String type, final String id) {
+        LOGGER.trace("Checking for existance of document: '{}/{}/{}'", index, type, id);
+        GetRequestBuilder builder = client.prepareGet()
+            .setIndex(index)
+            .setType(type)
+            .setId(id)
+            .setFields(new String[] {});
+        addCommonHeaders(builder);
+        GetResponse response = builder.get();
+        final boolean exists = response.isExists();
+        LOGGER.trace("Document '{}/{}/{}' exists? {}", index, type, id, exists);
+        return exists;
+    }
+    
+    /**
+     * Retrieve the set of indices for a given alias
+     * 
+     * @param alias The alias to lookup
+     * @return The set of indices to the given alias
+     */
+    public Set<String> getIndicesForAlias(String alias){
+        LOGGER.trace("Retrieving indices for alias '{}'", alias);
+        GetAliasesRequestBuilder builder = this.client.admin().indices().prepareGetAliases(alias);
+        addCommonHeaders(builder);
+        GetAliasesResponse response = builder.get();
```

Comment:
What happens if `alias` doesn't exist?  Will `builder.get()` return a `404`?  Will it throw an exception?  Will `response.getAliases().keysIt()` return `null`?  an "empty" iterator?

Reply:
An empty iterator is returned.